### DISCUSSION TITLE
Pyright specific ignore

### DISCUSSION THREAD

**Discussion by Pyprohly** [microsoft/pyright#2477] [Q&A]
Is there a Pyright specific variant of `# type: ignore`?

Also, how can I suppress a warning on a line.

**Comment by erictraut** [comment]
Pyright supports `# type: ignore` comments. This is specified within PEP 484, so all type checkers are expected to support it.

Pyright also allows you to enable or disable specific diagnostic rules within on a per-file basis by adding a comment at the top of the file: `# pyright: reportUnusedImport=false`.

**Reply by Pyprohly** [reply]
Mypy has a `--warn-unused-ignores` command-line option. If I start to introduce `# type: ignore`s that are only intended for Pyright, the Mypy `--warn-unused-ignores` option becomes less potent. So far I’ve been literally ignoring the errors Pyright produces because of no multi-pass support.

I wish `# pyright: reportGeneralTypeIssues=false` worked on a per-line basis.

**Reply by erictraut** [reply]
We're unlikely to invest in a non-standard way to suppress errors on a per-line basis when a standard way already exists.

For what it's worth, I'm not a fan of `# type: ignore` comments or other mechanisms to suppress an error on a single line. There's even an option in pyright to disable type ignores (`enableTypeIgnoreComments=false`), and I use that in my team's code base.

In my experience, there's always a better way to address a type error. Inevitably, a `# type: ignore` becomes maintenance issue that masks real bugs over time. 

I realize that it's tempting to use them because they require little or no thought. My advice is to do the little extra work involved in actually addressing the error and eliminate the need for a `# type: ignore`.

**Reply by guillaume-mueller** [reply]
What would be the relevant little extra work when using wrongly documented functions such as OpenCV's ones?
